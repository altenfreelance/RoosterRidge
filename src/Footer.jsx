import React, { Component } from "react";
import "./footer.css";


export default class Footer extends Component {
  render() {
    return (
      <footer className="mastfoot mt-auto">
        <section id="lab_social_icon_footer">
          <link
            href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
            rel="stylesheet"
          />
          <div className="container">
            <div className="text-center center-block">
            <a href="https://www.youtube.com/channel/UCE0jXwLwL7T6zIoQCnHVUzw">
                <i
                  id="social-yt"
                  className="fa fa-youtube-square fa-3x social"
                ></i>
              </a>
              <a href="https://www.facebook.com/RoosterRidgeOfficial/">
                <i
                  id="social-fb"
                  className="fa fa-facebook-square fa-3x social"
                ></i>
              </a>
              <a href="https://www.instagram.com/roosterridgeofficial/">
                <i id="social-ig" className="fa fa-instagram fa-3x social"></i>
              </a>
              <a href="mailto:roosterridgeofficial@gmail.com">
                <i
                  id="social-em"
                  className="fa fa-envelope-square fa-3x social"
                ></i>
              </a>
            </div>
          </div>
        </section>
      </footer>
    );
  }
}
