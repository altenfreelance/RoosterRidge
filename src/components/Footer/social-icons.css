.social:hover {
  -webkit-transform: scale(1.1);
  -moz-transform: scale(1.1);
  -o-transform: scale(1.1);
}

a .social {
  color: white
}

.social {
  -webkit-transform: scale(0.8);
  /* Browser Variations: */

  -moz-transform: scale(0.8);
  -o-transform: scale(0.8);
  -webkit-transition-duration: 0.5s;
  -moz-transition-duration: 0.5s;
  -o-transition-duration: 0.5s;
}

/*
      Multicoloured Hover Variations
  */

#social-bandcamp:hover {
  color: #0687f5;
}

#social-spotify:hover {
  color: #1DB954;
}

#social-apple:hover {
  color: #fa586a;
}

#social-yt:hover {
  color: #ff0000;
}

#social-fb:hover {
  color: #3B5998;
}

#social-ig:hover {
  color: #d6249f;
}

#social-em:hover {
  color: #f39c12;
}