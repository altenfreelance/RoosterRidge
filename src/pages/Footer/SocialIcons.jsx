import React from 'react';
import "./social-icons.css";


export default function SocialIcons() {
    return (
        <section>
            <link
                href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css"
                rel="stylesheet"
            />
            <div className="container">
                <div className="text-center center-block">
                    <a href="https://open.spotify.com/artist/7F5waAovJLtcJMpMM4C7yx?si=EhbY9WyNR2CJ5evtQi_4KA">
                        <i
                            title="Rooster Ridge on Spotify"
                            id="social-spotify"
                            className="fa fa-brands fa-spotify social fa-3x">
                        </i>
                    </a>
                    <a href="https://music.apple.com/us/artist/rooster-ridge/1618610103">
                        <i
                            title="Rooster Ridge on Apple Music"
                            id="social-apple"
                            className="fa fa-brands fa-apple social fa-3x">
                        </i>
                    </a>
                    <a href="https://roosterridge.bandcamp.com/releases">
                        <i
                            title="Rooster Ridge on Bandcamp"
                            id="social-bandcamp"
                            className="fa fa-brands fa-bandcamp social fa-3x">
                        </i>
                    </a>
                    <a href="https://www.youtube.com/channel/UCE0jXwLwL7T6zIoQCnHVUzw">
                        <i
                            title="Rooster Ridge on Youtube"
                            id="social-yt"
                            className="fa fa-brands fa-youtube fa-3x social"
                        ></i>
                    </a>
                    <a href="https://www.facebook.com/RoosterRidgeOfficial/">
                        <i
                            title="Rooster Ridge on Facebook"
                            id="social-fb"
                            className="fa fa-brands fa-facebook fa-3x social"
                        ></i>
                    </a>
                    <a href="https://www.instagram.com/roosterridgeofficial/">
                        <i
                            title="Rooster Ridge on Instagram"
                            id="social-ig"
                            className="fa fa-brands fa-instagram fa-3x social"></i>
                    </a>
                    <a href="mailto:roosterridgeofficial@gmail.com">
                        <i
                            title="Email Rooster Ridge"
                            id="social-em"
                            className="fa fa-envelope-square fa-3x social"
                        ></i>
                    </a>
                </div>
            </div>
        </section>
    )
}

