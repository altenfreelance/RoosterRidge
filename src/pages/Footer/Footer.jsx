import React, { Component } from "react";
import SocialIcons from "./SocialIcons";


export default class Footer extends Component {
  render() {
    return (
      <footer className="mastfoot mt-auto"
        style={{
          backgroundColor: 'rgba(128,0,0, 0.8)',
          position: 'absolute',
          bottom: '0',
          left: '0',
          hieght: '2.5rem',
          right: '0'
        }}>
        <SocialIcons />
      </footer>
    );
  }
}
