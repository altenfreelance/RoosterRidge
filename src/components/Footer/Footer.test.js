import React from 'react';
import Footer from './Footer';

import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders (matches snapshot) ', () => {
    const {asFragment} = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
});
