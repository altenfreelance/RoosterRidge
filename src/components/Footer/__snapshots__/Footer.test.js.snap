// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (matches snapshot)  1`] = `
<DocumentFragment>
  <footer
    class="mastfoot mt-auto"
    style="background-color: rgba(128, 0, 0, 0.8); position: absolute; bottom: 0px; left: 0px; right: 0px;"
  >
    <section>
      <link
        href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css"
        rel="stylesheet"
      />
      <div
        class="container"
      >
        <div
          class="text-center center-block"
        >
          <a
            href="https://open.spotify.com/artist/7F5waAovJLtcJMpMM4C7yx?si=EhbY9WyNR2CJ5evtQi_4KA"
          >
            <i
              class="fa fa-brands fa-spotify social fa-3x"
              id="social-spotify"
              title="Rooster Ridge on Spotify"
            />
          </a>
          <a
            href="https://music.apple.com/us/artist/rooster-ridge/1618610103"
          >
            <i
              class="fa fa-brands fa-apple social fa-3x"
              id="social-apple"
              title="Rooster Ridge on Apple Music"
            />
          </a>
          <a
            href="https://roosterridge.bandcamp.com/releases"
          >
            <i
              class="fa fa-brands fa-bandcamp social fa-3x"
              id="social-bandcamp"
              title="Rooster Ridge on Bandcamp"
            />
          </a>
          <a
            href="https://www.youtube.com/channel/UCE0jXwLwL7T6zIoQCnHVUzw"
          >
            <i
              class="fa fa-brands fa-youtube fa-3x social"
              id="social-yt"
              title="Rooster Ridge on Youtube"
            />
          </a>
          <a
            href="https://www.facebook.com/RoosterRidgeOfficial/"
          >
            <i
              class="fa fa-brands fa-facebook fa-3x social"
              id="social-fb"
              title="Rooster Ridge on Facebook"
            />
          </a>
          <a
            href="https://www.instagram.com/roosterridgeofficial/"
          >
            <i
              class="fa fa-brands fa-instagram fa-3x social"
              id="social-ig"
            />
          </a>
          <a
            href="mailto:roosterridgeofficial@gmail.com"
          >
            <i
              class="fa fa-envelope-square fa-3x social"
              id="social-em"
              title="Email Rooster Ridge"
            />
          </a>
        </div>
      </div>
    </section>
  </footer>
</DocumentFragment>
`;
